databricks-sdk-java/src/main/java/com/databricks/sdk/core/utils/OSUtils.java

```diff
@@ -1,44 +1,74 @@
  */
-public interface OSUtils {
+public class OSUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(OSUtils.class);
 
   /**
    * Returns the name of the current operating system.
    *
    * @return a String representing the name of the current operating system. The value will be "win"
    *     for Windows, "mac" for macOS, and "linux" for Linux-based operating systems.
    */
-  default String getOS() {
-    if (System.getProperty("os.name").toLowerCase().startsWith("win")) {
+  static String getOS(Environment env) {
+    String systemName = env.getSystemName().toLowerCase();
+    if (systemName.startsWith("win")) {
       return "win";
-    } else if (System.getProperty("os.name").toLowerCase().startsWith("mac")) {
+    } else if (systemName.startsWith("mac")) {
       return "mac";
     }
     return "linux";
   }
 
-  /**
-   * Returns a list of strings representing an executable command for the current operating system,
-   * based on the given list of command tokens.
-   *
-   * @param cmd a list of strings representing the command to be executed.
-   * @return a List of strings representing the executable command for the current operating system.
-   *     On Windows, the command will be wrapped in "cmd.exe /c" and on other operating systems, it
-   *     will be wrapped in "/bin/bash -c".
-   */
-  default List<String> getCliExecutableCommand(List<String> cmd) {
-    String cmdToString = String.join(" ", cmd);
-    if (getOS().equals("win")) {
-      return Arrays.asList("cmd.exe", "/c", cmdToString);
-    } else {
-      return Arrays.asList("/bin/bash", "-c", cmdToString);
+  /** Returns the OS-specific utilities for the current operating system. */
+  public static OSUtilities get(Environment env) {
+    String os = getOS(env);
+    switch (os) {
+      case "win":
+        return new WindowsUtilities(env);
+      case "mac":
+        return new MacOSUtilities(env);
+      case "linux":
+        return new LinuxUtilities(env);
+    }
+    throw new DatabricksException("Unsupported OS: " + os);
+  }
+
+  public static String findExecutable(List<String> paths, String name) {
+    LOG.debug("Searching for executable named '" + name + "' in : " + String.join(", ", paths));
+    for (String dir : paths) {
+      Path path = Paths.get(dir, name).toAbsolutePath().normalize();
+      if (!Files.isRegularFile(path)) {
+        continue;
+      }
+      long size;
+      try {
+        size = Files.size(path);
+      } catch (IOException e) {
+        LOG.debug("Unable to get size of databricks cli, continuing", e);
+        continue;
+      }
+      if (size < 1024 * 1024) {
+        LOG.debug("Databricks CLI version <0.100.0 detected");
+        continue;
+      }
+      String pathStr = path.toString();
+      LOG.debug("Found executable named '" + name + "' in PATH: " + pathStr);
+      return pathStr;
     }
+    LOG.debug("Failed to find executable named '" + name + "' in PATH");
```

Comment:
Since we are mentioning failed in the message -- this probably should be LOG.warn

Reply:
I don't think this is an indication that there is a problem though. The user may legitimately not have installed the CLI in their system (e.g. on a webserver).